utils/update-checkout

```diff
@@ -34,56 +38,53 @@ sys.path.append(os.path.join(SCRIPT_DIR, 'swift_build_support'))
 from swift_build_support import shell  # noqa (E402)
 
 
-def update_single_repository(repo_path, branch, reset_to_remote, should_clean,
-                             cross_repo):
+def update_single_repository(args):
```

Comment:
``Pool.map_async`` has to call ``update_single_repository`` with the fixed five arguments. I unpack them on the second line, maybe that's not as pythonic as it could be? ``args`` could be a tuple instead of a list as well.